src/utils/dateUtils.ts

```diff
@@ -71,6 +71,10 @@ export function parseDateString(dateString: string): Date | null {
   let year = dateString.slice(4);
   const today = new Date();
 
+  if (year.length !== 2 && year.length !== 4) {
+    return null;
+  }
```

Comment:
Hadde det lest bedre om det sto `if !(year.length === 2 || year.length === 4)`, altså at hvis lengden ikke er to eller fire så gir vi `null` tilbake.